### PR TITLE
chore: adds variant tracking attribute / event

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksadds-variants-heap-track-hashicorp.vercel.app) 🔎
- [Asana task](https://app.asana.com/0/1204333057896641/1204621999504620) 🎟️

## 🗒️ What

Adds analytics configuration for variants. Specifically, adds a custom `data-heap-track` attribute and fires off an event when a variant is selected. 

## 🧪 Testing

- Visit Segment > Sources > Staging & Local Sites > Debugger
- [Visit the preview ](https://dev-portal-git-ksadds-variants-heap-track-hashicorp.vercel.app/consul/tutorials/kubernetes/kubernetes-api-gateway)with a variant, select a variant option from the meta list or sidecar dropdown. 
- Validate that the 'Variant Selected event appears in the segment debugger feed. 
Screenshot 2023-06-14 at 11 49 10 AM

### CODE REVIEW

**Comment by kendallstrautman** [comment]
@BrandonRomano a starting point for my interpretation of what Tony needs for tracking. To track the active variant element, he could target `aria-current` within the `variant-list` container.

File: src/views/tutorial-view/components/variant-dropdown-disclosure/index.tsx [hunk 1]
```diff
@@ -53,6 +54,10 @@ export function VariantDropdownDisclosure({
 								)}
 								onClick={() => {
 									handleVariantCookie(variant.slug, option.slug)
+									safeAnalyticsTrack('Variant Selected', {
+										variant: variant.slug,
+										option: option.slug,
```
Comment (kendallstrautman) [inline src/views/tutorial-view/components/variant-dropdown-disclosure/index.tsx:59]: should probably add the path here

File: src/components/tutorial-meta/components/variant-list/index.tsx [hunk 2]
```diff
@@ -17,10 +17,13 @@ export function VariantList() {
...
 		<>
-			<div className={s.desktopVariantList}>
+			<div className={s.desktopVariantList} data-heap-track="variant-list">
 				<DesktopVariantList variant={currentVariant} />
 			</div>
-			<div className={s.mobileVariantDropdownDisclosure}>
+			<div
+				className={s.mobileVariantDropdownDisclosure}
+				data-heap-track="variant-list"
+			>
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/variant-list/index.tsx:26]: Will validate this in heap after its merged. Only one of these shows at a time depending on viewport width.

**Review by BRKalow — APPROVED** [review]
lgtm

File: src/components/tutorial-meta/components/variant-list/index.tsx (hunk 2 above)
Comment (kendallstrautman) [reply src/components/tutorial-meta/components/variant-list/index.tsx:26]: 💭 actually im going to remove one, we don't need both

**Pull request merged**